src/smart-components/group/groups.js

```diff
@@ -176,10 +179,24 @@ const Groups = () => {
         ]
       : []),
   ];
+
   const data = groups.map((group) =>
     group.platform_default || group.admin_default ? { ...group, principalCount: `All${group.admin_default ? ' org admins' : ''}` } : group
   );
-  const rows = createRows(isAdmin, data, selectedRows);
+
+  const fetchExpandTableRoleDetailsForGroup = (uuid) => dispatch(fetchRolesForGroup(uuid, defaultCompactSettings));
+  const fetchExpandTableMemberDetailsForGroup = (uuid) => dispatch(fetchMembersForGroup(uuid, defaultCompactSettings));
+
+  const onExpand = (_event, _rowIndex, colIndex, isOpen, rowData) => {
+    if (!isOpen) {
+      setExpanded({ ...expanded, [rowData.uuid]: colIndex + Number(!isAdmin) });
+      colIndex + Number(!isAdmin) === 2 && fetchExpandTableRoleDetailsForGroup(rowData.uuid) && fetchExpandTableMemberDetailsForGroup(rowData.uuid);
```

Comment:
we probably want to fetch roles data for index === 2 and members for index === 3. Now it is fetching both for the roles cell